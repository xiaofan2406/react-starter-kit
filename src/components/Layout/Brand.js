import React from 'react';
import { css } from 'emotion/react';
import { variables, fontSizes, theme } from 'styles';
import { APP_TITLE } from 'configs';
import logo from 'assets/logo.svg';

const brand = css`
  display: flex;
  align-items: center;
  img {
    animation: spin infinite 10s linear;
    height: calc(${variables.Brand.logoHeight} * 1px);
  }
  span {
    animation: fadeIn 2s ease;
    font-size: calc(${fontSizes.large} * 1px);
    color: ${theme.inverseColor};
  }
`;

class Brand extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <div className={brand}>
        <img src={logo} alt="logo" />
        <span>
          {APP_TITLE}
        </span>
      </div>
    );
  }
}

export { Brand as Component };

export default Brand;
