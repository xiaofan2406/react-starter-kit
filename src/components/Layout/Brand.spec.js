import React from 'react';
import renderer from 'react-test-renderer';

import Brand from './Brand';

it('should match snapshot', () => {
  const component = renderer.create(<Brand />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
