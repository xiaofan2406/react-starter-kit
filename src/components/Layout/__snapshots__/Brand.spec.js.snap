// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="css-brand-2sf3o9 vars-123jc49"
>
  <img
    alt="logo"
    src="logo.svg"
  />
  <span>
    React Starter Kit
  </span>
</div>
`;
